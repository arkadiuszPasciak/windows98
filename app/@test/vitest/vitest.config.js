import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
