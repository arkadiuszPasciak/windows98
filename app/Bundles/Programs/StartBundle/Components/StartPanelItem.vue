<template>
  <div
    class="StartPanelItem"
    :class="`s-${size}`"
    @click="$emit('openProgram')"
  >
    <img
      class="icon"
      :width="size === 'medium' ? 25 : 18"
      :height="size === 'medium' ? 25 : 18"
      :src="`/icons/applications/${iconName}.png`"
    />
    <UIText class="name">{{ t(name) }}</UIText>

    <div v-if="$slots.default" class="under-panel"><slot></slot></div>
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIText from '@APP|Bundles/UITextBundle/Components/UIText.vue'

  defineProps({
    size: {
      type: String as PropType<'small' | 'medium'>,
    },
    name: {
      type: String,
      required: true,
    },
    iconName: {
      type: String,
      required: true,
    },
  })

  const { t } = useI18n()

  defineEmits(['openProgram'])
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/StartBundle/Styles/StartPanelItem.styles.scss"
/>
