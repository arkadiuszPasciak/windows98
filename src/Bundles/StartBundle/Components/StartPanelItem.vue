<template>
  <div class="StartPanelItem" :class="`s-${size}`">
    <img
      class="icon"
      :width="size === 'medium' ? 25 : 18"
      :height="size === 'medium' ? 25 : 18"
      :src="`src/Assets/Icons/Applications/${iconName}.png`"
    />
    <UIText>{{ t(name) }}</UIText>

    <div v-if="$slots.default" class="under-panel"><slot></slot></div>
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIText from '@Bundles/UIBundle/Components/UIText.vue'

  defineProps({
    size: {
      type: String as PropType<'small' | 'medium'>,
    },
    name: {
      type: String,
      required: true,
    },
    iconName: {
      type: String,
      required: true,
    },
  })

  const { t } = useI18n()
</script>

<i18n src="@Bundles/StartBundle/Locales/Start.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/StartBundle/Styles/StartPanelItem.styles.scss';
</style>
