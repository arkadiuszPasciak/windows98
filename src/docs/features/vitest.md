# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
