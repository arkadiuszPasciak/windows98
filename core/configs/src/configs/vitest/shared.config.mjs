import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
