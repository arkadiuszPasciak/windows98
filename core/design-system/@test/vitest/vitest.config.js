import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
