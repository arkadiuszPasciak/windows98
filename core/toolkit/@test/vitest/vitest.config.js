import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
