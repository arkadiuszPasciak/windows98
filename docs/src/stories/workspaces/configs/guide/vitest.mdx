import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

