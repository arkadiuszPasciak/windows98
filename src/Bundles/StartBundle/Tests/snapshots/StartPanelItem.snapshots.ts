import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import StartPanelItem from '@Bundles/StartBundle/Components/StartPanelItem.vue'

const sizeSmallWrapper = mount(StartPanelItem, {
  props: {
    name: 'StartBundle.shut-down',
    iconName: 'shut-down',
    size: 'small',
  },
})

const sizeMediumWrapper = mount(StartPanelItem, {
  props: {
    name: 'StartBundle.documents',
    iconName: 'documents',
    size: 'medium',
  },
})

describe('[StartBundle]<Snapshots>(StartPanelItem)', async () => {
  it('renders correctly size small component', () => {
    expect(sizeSmallWrapper.element).toMatchSnapshot()
  })

  it('renders correctly size medium component', () => {
    expect(sizeMediumWrapper.element).toMatchSnapshot()
  })
})
