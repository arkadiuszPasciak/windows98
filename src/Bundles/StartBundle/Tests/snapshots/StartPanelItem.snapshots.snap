// Vitest Snapshot v1

exports[`[StartBundle]<Snapshots>(StartPanelItem) > renders correctly size medium component 1`] = `
<div
  class="StartPanelItem s-medium"
  data-v-f765ff07=""
>
  <img
    class="icon"
    data-v-f765ff07=""
    height="25"
    src="/icons/applications/documents.png"
    width="25"
  />
  <p
    class="UIText name"
    data-v-f0b878d5=""
    data-v-f765ff07=""
  >
    
    Documents
    
  </p>
  <!--v-if-->
</div>
`;

exports[`[StartBundle]<Snapshots>(StartPanelItem) > renders correctly size small component 1`] = `
<div
  class="StartPanelItem s-small"
  data-v-f765ff07=""
>
  <img
    class="icon"
    data-v-f765ff07=""
    height="18"
    src="/icons/applications/shut-down.png"
    width="18"
  />
  <p
    class="UIText name"
    data-v-f0b878d5=""
    data-v-f765ff07=""
  >
    
    Shut Down...
    
  </p>
  <!--v-if-->
</div>
`;
